lib/active_merchant/billing/gateways/paymentez.rb

```diff
@@ -134,6 +134,9 @@ def add_customer_data(post, options)
         post[:user][:email] = options[:email]
         post[:user][:ip_address] = options[:ip] if options[:ip]
         post[:user][:fiscal_number] = options[:fiscal_number] if options[:fiscal_number]
+        if phone = options[:phone] || (options[:billing_address][:phone] if options[:billing_address])
```

Comment:
I really don't like the `foo || (bar if baz)` pattern, because it's very unclear what actually happens. (You [responded `ew` to this same pattern](https://spreedly.slack.com/archives/C5GBHQTMH/p1537286552000100), BTW.). There's a couple routes here, but since we're on Ruby 2.3.6 now, I'd suggest `if phone = (options[:phone] || options[:billing_address]&.[](:phone))` or `phone = (options[:phone] || (options[:billing_address] || [])[:phone])`, or something similar to be a bit more explicit.

Reply:
Thanks, updated. I was not familiar with `&.`, perfect!

Reply:
Better than what I just said is to use `.dig`; sorry I spaced that. So that'd be `phone = options[:phone] || options.dig(:billing_address, :phone)`